client/reader_slice_of_slices_iterator.go

```diff
@@ -111,11 +114,34 @@ func (it *readerSliceOfSlicesIterator) CurrentLen() int {
 	return len(it.segments[it.idx].Unmerged)
 }
 
-func (it *readerSliceOfSlicesIterator) CurrentAt(idx int) io.Reader {
+func timeConvert(ticks *int64) time.Time {
+	if ticks == nil {
+		return time.Time{}
+	}
+	return xtime.FromNormalizedTime(*ticks, time.Second)
+}
+
+func (it *readerSliceOfSlicesIterator) CurrentStart() time.Time {
+	segments := it.segments[it.idx]
+	if segments.Merged != nil {
+		return timeConvert(segments.Merged.StartTime)
+	}
+	return timeConvert(segments.Unmerged[0].StartTime)
```

Comment:
Perhaps just make sure that `len(segments.Unmerged) > 0` before this to avoid nil ptr deref if there isn't an entry?

Reply:
:+1: